Flaky test: TestMRFailoverLeaderElection

### Current Behavior

TestMRFailoverLeaderElection that is implemented in `internal/metarepos/raft_metadata_repository_test.go` is failed sometimes.

```
=== RUN   TestMRFailoverLeaderElection

  Given MR cluster ✔✔✔✔✔✔✔✔✔
    When node fail ✔
      Then MR Cluster should elect ✔✘
GC Stat:: 63.151535 -> 7.620499 mb, Sys: 116.769554 mb



Failures:

  * /Users/ijsong/workspace/varlog/internal/metarepos/raft_metadata_repository_test.go
  Line 1549:
  Expected: true
  Actual:   false
  goroutine 972 [running]:
  github.daumkakao.com/varlog/varlog/internal/metarepos.TestMRFailoverLeaderElection.func1.4.1({0xc00000e030?, 0xc0021603f0?})
        /Users/ijsong/workspace/varlog/internal/metarepos/raft_metadata_repository_test.go:1549 +0x14b
  github.com/jtolds/gls.(*ContextManager).SetValues.func1(0x0)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:97 +0x5c4
  github.com/jtolds/gls.EnsureGoroutineId(0xc0021603c0)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/gid.go:19 +0x1a2
  github.com/jtolds/gls.(*ContextManager).SetValues(0xc0004152d0, 0xc002160360, 0xc00048e030)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:63 +0x29e
  github.daumkakao.com/varlog/varlog/internal/metarepos.TestMRFailoverLeaderElection.func1.4({0xc00000e030?, 0xc0022b2240?})
        /Users/ijsong/workspace/varlog/internal/metarepos/raft_metadata_repository_test.go:1542 +0x19c
  github.com/jtolds/gls.(*ContextManager).SetValues.func1(0x0)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:97 +0x5c4
  github.com/jtolds/gls.EnsureGoroutineId(0xc0022b2210)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/gid.go:19 +0x1a2
  github.com/jtolds/gls.(*ContextManager).SetValues(0xc0004152d0, 0xc0022b21b0, 0xc00207c630)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:63 +0x29e
  github.daumkakao.com/varlog/varlog/internal/metarepos.TestMRFailoverLeaderElection.func1({0xc00000e030?, 0xc001e46180?})
        /Users/ijsong/workspace/varlog/internal/metarepos/raft_metadata_repository_test.go:1537 +0xd56
  github.com/jtolds/gls.(*ContextManager).SetValues.func1(0x0)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:97 +0x5c4
  github.com/jtolds/gls.EnsureGoroutineId.func1()
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/gid.go:24 +0x3f
  github.com/jtolds/gls._m(0x0, 0xc00000e060)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/stack_tags.go:108 +0x42
  github.com/jtolds/gls.github_com_jtolds_gls_markS(0x57642a0?, 0xc001e460f0?)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/stack_tags.go:56 +0x37
  github.com/jtolds/gls.addStackTag(...)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/stack_tags.go:49
  github.com/jtolds/gls.EnsureGoroutineId(0xc001e46120?)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/gid.go:24 +0x15d
  github.com/jtolds/gls.(*ContextManager).SetValues(0xc0004152d0, 0xc001e460c0, 0xc002340020)
        /Users/ijsong/workspace/varlog/vendor/github.com/jtolds/gls/context.go:63 +0x29e
  github.daumkakao.com/varlog/varlog/internal/metarepos.TestMRFailoverLeaderElection(0xc001ba6680)
        /Users/ijsong/workspace/varlog/internal/metarepos/raft_metadata_repository_test.go:1490 +0xe8
  testing.tRunner(0xc001ba6680, 0x4fb9230)
        /usr/local/Cellar/go/1.18.1/libexec/src/testing/testing.go:1439 +0x214
  created by testing.(*T).Run
        /usr/local/Cellar/go/1.18.1/libexec/src/testing/testing.go:1486 +0x725



1710 total assertions

--- FAIL: TestMRFailoverLeaderElection (24.44s)
```

### Expected Behavior


### Steps To Reproduce

Unit test (e.g., `make test`)

### Environment

- Varlog version:
- OS:
- Architecture:

### Anything else

--- a/internal/metarepos/raft_metadata_repository.go
+++ b/internal/metarepos/raft_metadata_repository.go
@@ -135,6 +135,7 @@ func NewRaftMetadataRepository(options *MetadataRepositoryOptions) *RaftMetadata
 		sw:                stopwaiter.New(),
 		tmStub:            tmStub,
 		topicEndPos:       make(map[types.TopicID]int),
+		requestNum:        uint64(time.Now().UnixNano()),
 	}
 
 	mr.storage = NewMetadataStorage(mr.sendAck, options.SnapCount, mr.logger.Named("storage"))

--- a/internal/metarepos/storage.go
+++ b/internal/metarepos/storage.go
@@ -431,6 +431,11 @@ func (ms *MetadataStorage) registerLogStream(ls *varlogpb.LogStreamDescriptor) e
 		return verrors.ErrAlreadyExists
 	}
 
+	if equal {
+		// To ensure that it is applied to the meta cache
+		return nil
+	}
+
 	_, cur := ms.getStateMachine()
 
 	ms.mtMu.Lock()

